integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -189,16 +201,40 @@ def _create_connection(self):
         self._cursor = self._connection.cursor()
         self._dict_cursor = self._connection.cursor(row_factory=dict_row)
 
-        # Init schema
+        if not self._table_initialized:
+            self._initialize_table()
+
+        return self._connection
+
+    def _initialize_table(self):
+        """
+        Internal method to initialize the table.
+        """
         if self.recreate_table:
             self.delete_table()
+
         self._create_table_if_not_exists()
         self._create_keyword_index_if_not_exists()
 
         if self.search_strategy == "hnsw":
             self._handle_hnsw()
 
-        return self._connection
+        self._table_initialized = True
+
+    @staticmethod
+    def _check_connection(connection):
```

Comment:
I would rename this method to something else, it's not clear what check actually means at first glance.

Something like this probably:

```suggestion
    def _connection_is_valid(connection):
```

Reply:
right!

Done.